_posts/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that.md

```diff
@@ -0,0 +1,325 @@
+
+    // then
+    expect(screen.getByText('0')).toBeInTheDocument();
+  });
+  it('should render defaultCounter from props', () => {
+    // when
+    render(<Test defaultCounter={100} />);
+
+    // then
+    expect(screen.getByText('100')).toBeInTheDocument();
+  });
+  it('should increment conunter on click', () => {
+    // given
+    render(<Test defaultCounter={100} />);
+
+    const button = screen.getByRole('button', { name: 'Increment counter' });
+
+    // when
+    userEvent.click(button);
+
+    // then
+    expect(screen.getByText('101')).toBeInTheDocument();
+  });
+  it('should decrement conunter on click', () => {
+    // given
+    render(<Test defaultCounter={100} />);
+
+    const button = screen.getByRole('button', { name: 'Decrement counter' });
+
+    // when
+    userEvent.click(button);
+
+    // then
+    expect(screen.getByText('99')).toBeInTheDocument();
+  });
+});
+
+```
+## Enzyme and RTL in Allegro Ads
+
+My team is currently working on the development of [Allegro Ads](https://allegro.pl/ads), which is an advertising tool 
+for our ecommerce platform sellers. It is a whole panel for viewing statistics, displaying advertising campaigns. We 
+provide sponsored offers for our allegro site, facebook, google and recently even for on the partner websites. Our 
+sellers pay us for the opportunity to advertise their products, and at the same time we strive to ensure that these 
+ads accurately reach people potentially interested in buying, thereby generating profits for sellers.
+
+The main technology stack is standard and still pretty cool. The core of the project is written in React, and 
+significant parts have been rewritten to Typescript. From the beginning, for testing purposes, we use Jest. In 
+my opinion this is a very good and stable tool, easy to work with. It has quite a low entry threshold, even if someone 
+hasn’t had too much experience with tests before. The main advantages of Jest for me are minimal configuration and quite 
+good documentation. But for testing react components we currently have two libraries and I’ll explain why in a moment.
+
+The 2.0. version of platform Allegro Ads was released about 6 years ago, with an Enzyme library on a board. A year ago 
+the team added a react testing library to package.json. We have both packages so far. **Recently we officially decided 
+to slowly rewrite the Enzyme library for good.** We added the rule to our docc (Declaration of Code Convention) that 
+the new components are tested just in RTL, the old ones will be rewritten when existing parts of the code will be onboard.
+
+## Migration effects so far
+
+Currently we have got **592 test files** (counting after the file names with “spec” phrase). In these files we can 
+also find **709 of unique “describe”** usage for wrapping a series of tests, and **1796 unique test cases**. Half of 
+tests are written only in jest, but a significant amount of tests for better testing use extra libraries. 
```

Comment:
```suggestion
tests are written only in Jest, but a significant amount of tests for better testing use extra libraries. 
```
Szczególnie tutaj śmiesznie by to brzmiało 😄 [link](https://dictionary.cambridge.org/dictionary/english/jest?q=in+jest)
> If you say something **in jest**, you are not serious about it